modules/entity_version_history/src/Controller/EntityVersionHistoryController.php

```diff
@@ -62,17 +88,76 @@ public function historyOverview(RouteMatchInterface $route_match): array {
       $this->t('Date'),
       $this->t('Created by'),
     ];
-
     $rows = [];
 
+    /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+    $entity = $this->getEntityFromRouteMatch($route_match);
+    $langcode = $entity->language()->getId();
+    $entity_type_id = $entity->getEntityTypeId();
+    $entity_storage = $this->entityTypeManager->getStorage($entity_type_id);
+    $default_revision = $entity->getRevisionId();
+    $current_revision_displayed = FALSE;
+
+    // Get the version field name from the corresponding history config.
+    $history_storage = $this->entityTypeManager->getStorage('entity_version_history_settings');
+    $history_setting = $history_storage->load($entity_type_id . '.' . $entity->bundle());
+    $version_field = $history_setting->getTargetField();
+
+    foreach ($this->getRevisionIds($entity, $entity_storage) as $vid) {
+      /** @var \Drupal\Core\Entity\ContentEntityInterface $revision */
+      $revision = $entity_storage->loadRevision($vid);
+      // Only show revisions that are affected by the language that is being
+      // displayed at the moment.
+      if ($revision->hasTranslation($langcode) && $revision->getTranslation($langcode)->isRevisionTranslationAffected()) {
+        $version = $revision->get($version_field)->getValue();
```

Comment:
Let's check also there is a version in the field and print something if it's missing.